dataland-frontend/src/utils/SearchTaxonomyPageCompanyDataRequester.ts

```diff
@@ -64,13 +68,26 @@ export async function getCompanyDataForTaxonomyPage(
     | "Hdax"
     | "Dax50Esg",
   onlyCompanyNames: boolean,
+  frameworksToFilter: Array<string>,
   keycloakPromise: Promise<Keycloak>
 ): Promise<Array<object>> {
   let mappedResponse: object[] = [];
   try {
     const companyDataControllerApi = await new ApiClientProvider(keycloakPromise).getCompanyDataControllerApi();
     const response = await companyDataControllerApi.getCompanies(searchString, stockIndex, onlyCompanyNames);
-    mappedResponse = mapStoredCompanyToTaxonomyPage(response.data);
+    const responseData: Array<StoredCompany> = response.data;
+    let storedCompaniesToMapToPage;
+    if (frameworksToFilter && frameworksToFilter.length > 0) {
+      storedCompaniesToMapToPage = responseData.filter(
+        (storedCompany) =>
+          storedCompany.dataRegisteredByDataland.filter((dataMetaInfo) =>
+            frameworksToFilter.includes(dataMetaInfo.dataType)
+          ).length > 0
```

Comment:
Dieser API endpoint liefer ungefragt ziemlich viele Daten zurück oder nicht?

Reply:
Ja er liefert alle zurück die zum searchString über substring-matching passen.

Anschließend filtert er client-seitig auf evtl. zu filternde frameworks.

=> Bad performance, aber ich dachte mir erstmal eine akzeptable Lösung, bis wir die Filter-Methode im Backend implementieren.

Auch hier: Möchtest du ein Ticket erstellen, dass die Filter-Option ins BAckend implementiert und die Funktionalität vom Client-Code wieder wegnimmt?

Reply:
Das verstehe ich jetzt nicht ganz. Wenn das Frontend nach einer gefilterten Datenmenge fragt dann sollte das backend auch nur diese liefern und nicht alles zurückgeben und der client kann sich dann raussuchen was er gerne hätte. Bei der Option mit dem Index wird doch auch nur das zurückgegeben was in dem Index ist und nicht erstmal alles und der Client soll filtern. Die entsprechende Suche im Backend so anzupassen das es funktioniert würde ich auch ganz klar als Teil dieser Story sehen.

Reply:
Done